packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx

```diff
@@ -70,183 +76,245 @@ const RedshiftForm: FC<{
+                                        </p>
+                                    }
+                                    required
+                                    disabled={disabled}
+                                />
+                            )}
                         />
-                    )}
-                    <Button
-                        text={
-                            sshTunnelPublicKey
-                                ? `Regenerate key`
-                                : `Generate public key`
-                        }
-                        onClick={() => mutate()}
-                        loading={isLoading}
-                        disabled={disabled || isLoading}
-                    />
+                        <Controller
+                            name="warehouse.sslmode"
+                            defaultValue="prefer"
+                            render={({ field }) => (
+                                <Select
+                                    label="SSL mode"
+                                    description={
+                                        <p>
+                                            This controls how dbt connects to
+                                            Postgres databases using SSL. You
+                                            can see more details in{' '}
+                                            <Anchor
+                                                target="_blank"
+                                                href="https://docs.getdbt.com/reference/warehouse-profiles/postgres-profile#sslmode"
+                                                rel="noreferrer"
+                                            >
+                                                dbt documentation
+                                            </Anchor>
+                                            .
+                                        </p>
+                                    }
+                                    data={[
+                                        'disable',
+                                        'no-verify',
+                                        'allow',
+                                        'prefer',
+                                        'require',
+                                        'verify-ca',
+                                        'verify-full',
+                                    ].map((x) => ({ value: x, label: x }))}
+                                    value={field.value}
+                                    onChange={field.onChange}
+                                    disabled={disabled}
+                                />
+                            )}
+                        />
+
+                        <BooleanSwitch
+                            name="warehouse.ra3Node"
+                            label="Use RA3 node"
+                            labelHelp="Allow dbt to use cross-database-resources."
```

Comment:
just wanted to ask can we use below code in place of BooleanSwitch defined above <Controller name="warehouse.ra3Node" render={({ field }) => ( <Switch.Group label="Use RA3 node" description="Allow dbt to use cross-database-resources." value={field.value} onChange={field.onChange} > <Switch onLabel="Yes" offLabel="No" {...field} disabled={disabled} value="true" onChange={(isChecked) => field.onChange(isChecked) } /> </Switch.Group> )} />